backend/src/main/kotlin/fr/gouv/dgampa/rapportnav/infrastructure/api/bff/MissionController.kt

```diff
@@ -91,20 +111,49 @@ class MissionController(
         }
     }
 
+    /**
+     * Retrieves general information about a mission by its ID, including distance and fuel consumption.
+     *
+     * This function queries the general information for a mission based on the provided mission ID.
+     * The general information includes details such as the distance traveled and the fuel consumption for the mission.
+     *
+     * @param missionId The ID of the mission for which to retrieve general information.
+     * @return The general information of the mission, including distance and fuel consumption, or null if no such information exists.
+     */
     @QueryMapping
     fun missionGeneralInfo(@Argument missionId: Int): MissionGeneralInfo? {
         val info = getMissionGeneralInfoByMissionId.execute(missionId)
             ?.let { MissionGeneralInfo.fromMissionGeneralInfoEntity(it) }
         return info
     }
 
+    /**
+     * Adds general information to a Mission object, including distance and fuel consumption.
+     *
+     * This function retrieves the general information for a given mission based on the mission's ID.
+     * The general information includes details such as the distance traveled and fuel consumption for the mission,
+     * and it adds this information to the Mission object.
+     *
+     * @param mission The mission object for which to retrieve and add general information.
+     * @return The general information of the mission, including distance and fuel consumption, or null if no such information exists.
+     */
     @SchemaMapping(typeName = "Mission", field = "generalInfo")
     fun missionGeneralInfoForMission(mission: Mission): MissionGeneralInfo? {
         val info = getMissionGeneralInfoByMissionId.execute(mission.id)
             ?.let { MissionGeneralInfo.fromMissionGeneralInfoEntity(it) }
         return info
     }
 
+    /**
+     * Updates the general information of a mission, including distance and fuel consumption.
```

Comment:
Wow c'est générer? ou t'as fait à la main?  t'es chaud :D

Reply:
GPT for the win